compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -384,7 +384,12 @@ class ComposeScene internal constructor(
      */
     var constraints: Constraints = Constraints()
         set(value) {
-            field = value
+            field = Constraints(
```

Comment:
I don't see any useful application of setting `minWidth`, so we probably should go with p.1 (and don't forget to do the same in ImageComposeScene)

Reply:
Just noticed: this PR breaks popups - now they fill the entire screen too. I'll try to fix.

Reply:
It looks like minWidth and minHeight are still needed for popups. When a popup gets attached to the scene, it would take the scene's constraints (or size if we choose to get rid of constraints). 
We want the root layout to fill the screen width/height. At the same time we want popups to have an appropriate size (not an entire screen), therefore we have to constrain them at some point (for example when it gets attached to the scene).
 
 Then later the given constraints are used in `measureAndLayout`:
 ```kotlin
 measureAndLayoutDelegate.updateRootConstraints(constraints)
 ```
 Which is a common api and I believe it needs constraints (size won't be enough) to measure LayoutNodes.

Reply:
One of the options is to apply different constraints for main owner and for popup owners. But I am not sure yet, if that is okay.

Also, I wonder, will this shows a normal Button, not stretched to the Window, if we implement new constraints?

```
singleWindowApplication {
  Button({}) {}
}
```

Reply:
> One of the options is to apply different constraints for main owner and for popup owners
yes, it was implemented in the last commit.

> Also, I wonder, will this shows a normal Button, not stretched to the Window, if we implement new constraints?
It's stretched. Same on Android by the way (when the button is a root node). It's weird a bit. 

What do you think about wrapping the content into an implicit container, for example, Box with Modifier.fillSize()? It would solve all the issues, I think (unless it would add some unexpected issues).

Reply:
```
 Box with Modifier.fillSize()
```
That will break the adaptive window feature (when the window size is equal content size).

> It's stretched. Same on Android by the way

Maybe, it is not a big deal 🤔. No one places simple components on the root level. It is always a container.

Reply:
Since android has it this way, I think on desktop it may behave similarly.

Beside that nuance, do you think this PR is okay?

Reply:
Yes, let's merge it :)

Reply:
wait )

it seems if we want to create a window with button, and make its size as the size of the button, we want be able to do that

Reply:
I don't see the way, how can we can keep adaptive size feature if we remove minWidth constraint :(

Reply:
Perhaps, we can add `Box(fillMaxSize)`, and make `contentSize` pick not the root components, but the children of this Box

Reply:
Do you mean that minWidth should be 0? 
Do we have an example with adaptive size, so I understand the issue better?

Reply:
> Do you mean that minWidth should be 0

Perhaps, I am not sure. May be it will work if we implement it in a way as I described in the latest message.

```
singleWindowApplication(state = WindowState(size = DpSize.Unspecified), undecorated = true) {
   Button({}) {}
}
```